language-support/java/codegen/src/main/scala/com/digitalasset/daml/lf/codegen/backend/java/inner/InterfaceClass.scala

```diff
@@ -0,0 +1,114 @@
+// Copyright (c) 2022 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package com.daml.lf.codegen.backend.java.inner
+
+import com.daml.ledger.javaapi
+import com.daml.lf.codegen.backend.java.JavaEscaper
+import com.daml.lf.data.Ref.{ChoiceName, PackageId, QualifiedName}
+import com.daml.lf.iface.{DefInterface, TemplateChoice}
+import com.squareup.javapoet._
+import com.typesafe.scalalogging.StrictLogging
+
+import javax.lang.model.element.Modifier
+
+object InterfaceClass extends StrictLogging {
```

Comment:
If you have duplicate definitions between here and `TemplateClass` consider pulling them out and calling them.